### PR TITLE
HTTP302FromOrigin

### LINKED ISSUE #12: feat: http302 from origin extension
After discussing with @RicoToothless, we are planing to implement a extension construct for [http302 from origin](https://github.com/awslabs/aws-cloudfront-extensions/tree/main/edge/nodejs/http302-from-origin).

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by RicoToothless**
This PR implements the [HTTP302 from location extension](https://github.com/awslabs/aws-cloudfront-extensions/tree/main/edge/nodejs/http302-from-origin) originally from SAR. Due to https://github.com/awslabs/aws-cloudfront-extensions/issues/109, we implement the lambda locally with TypeScript.

Fixes #12

### CODE REVIEW

File: src/demo/http302-from-origin/index.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+// create the cloudfront distribution with extension(s)
+const http302FromOrigin = new extensions.HTTP302FromOrigin(stack, 'http302');
+
+const demoAlbService = new ecsPatterns.ApplicationLoadBalancedFargateService(stack, 'DemoAlbService', {
+  taskImageOptions: { image: ecs.ContainerImage.fromRegistry('mendhak/http-https-echo:14') },
+  publicLoadBalancer: true,
+});
+
+// demoAlbService.listener.addRedirectResponse('302RedirectRule', {
```
Comment (guan840912) [inline src/demo/http302-from-origin/index.ts:19]: I think, she try to modify alb  listener to 302
cc @pahud

**Comment by pahud** [comment]
@RicoToothless @guan840912 please review

File: .projenrc.js [hunk 2]
```diff
@@ -24,12 +27,14 @@ const project = new AwsCdkConstructLibrary({
     '@aws-cdk/aws-sam',
     '@aws-cdk/aws-s3',
     '@aws-cdk/aws-s3-deployment',
+    '@aws-cdk/aws-ec2',
```
Comment (pahud) [inline .projenrc.js:30]: we probably can remove this.

File: lambda-assets/extensions/cf-http302-from-origin/index.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      body: newResponse.data,
+    }
+    return responseObject
+  } else {
+    return response;
+  }
+};
+
+
+function cacheControl(responseObject: HttpResponse,  maxAge: string) {
```
Comment (pahud) [inline lambda-assets/extensions/cf-http302-from-origin/index.ts:41]: will try leverage this function for easier custom cache control

**Review by guan840912 — APPROVED** [review]
ＬＧＴＭ

**Review by pahud — APPROVED** [review]

**Pull request merged**